Add binary generation for releases (#47)
observatorium/obsctl: docs/release.md

We use semantic versioning for `obsctl` in format `vX.Y.Z`.

> We have started including binaries after initial release. Users can also use `go install` to install obsctl.

Reviewer: ```suggestion
We also ship binaries <link to readme>. Users can also use `go install` to install obsctl.
```

Passive voice, we should probably just link to instructions on how to get the binary.

Author: Yup, changed, I'll update the readme install section once this change is in, and we have binaries.
